src/conways-game-of-life/index.ts

```diff
@@ -0,0 +1,81 @@
+const updateCellStatus = (x: number, y: number, currentCellStatus: number, neighbourSum: number) => {
+    if (currentCellStatus === 1 && neighbourSum > 3) {
+        return 0;
+    }
+    else if (currentCellStatus === 1 && neighbourSum < 2) {
+        return 0;
+    }
+    else if (currentCellStatus === 0 && neighbourSum === 3) {
+        return 1;
+    }
+    else {
+        return currentCellStatus;
+    }
+} //Apply the conditions for updating the status of a cell 
+
+const calculateNeighbourSum = (grid: number[][], x: number, y: number) => {
+    let toBeSummed: number[] = [
+        grid[y-1]?.[x+1],
+        grid[y]?.[x+1],
+        grid[y+1]?.[x+1],
+        grid[y+1]?.[x],
+        grid[y+1]?.[x-1],
+        grid[y]?.[x-1],
+        grid[y-1]?.[x-1],
+        grid[y-1]?.[x]   
+    ];
+    const toBeSummedClean = toBeSummed.filter(value => value !== undefined);
+    let neighbourSum = 0;
+    for (const value of toBeSummedClean) {
+        neighbourSum += value;    
+    }
+    return neighbourSum;    
+} // Calculates and return the sum of neighbors alive around the cell of interest
+
+
+const calculateNextGeneration = (grid: number[][]) => {
+    let nextGeneration: number[][] = grid.map(row => [...row]);
+    grid.forEach((row, rowIndex) => {
+        row.forEach((value, colIndex) => {
+            const currentCellStatus = value;
+            const y = rowIndex;
+            const x = colIndex;
+            const neighbourSum: number = calculateNeighbourSum(grid, x, y);
+            const newCellStatus: number = updateCellStatus(x, y, currentCellStatus, neighbourSum);
+            nextGeneration[y][x] = newCellStatus;
+        });
+    });
+    return nextGeneration;
+}  // Calculates the next generation using the returns of functions calculateNeighbourSum() and updateCellStatus()
+
+function sleep(time: number): Promise<void> {
+    return new Promise(resolve => setTimeout(resolve, time));
+}
```

Comment:
You used arrow functions in the whole code. Why change it here?

You did
```
function sleep(time: number): Promise<void> { ... }
```
Instead of
```
const sleep = (time: number) => { ... }
```

Safe to say that it's a copy and past from internet. lol

Reply:
It is. I still don't get how this function works and I had a hard time to implement it myself, so google helped me on that. I knew I could have used it after I saw your e-mail, but I still don't get the asynchronous function thing, even after reading about it and asking the AI to explain it to me.